examples/example-tls/src/main/java/io/grpc/examples/helloworldtls/HelloWorldServerS2A.java

```diff
@@ -0,0 +1,141 @@
+import io.netty.handler.ssl.SslProvider;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.logging.Logger;
+
+// TODO: change flags to use com.google.common.flags
+
+/**
+ * Server that manages startup/shutdown of a {@code Greeter} server with TLS 1.3 enabled.
+ *
+ * args:
+ * [port]
+ * [certChainFilePath] File Path to Server Certificate
+ * [privateKeyFilePath] File Path to Server Private Key
+ * [trustCertCollectionFilePath] File Path to CA Certificate
+ */
+public class HelloWorldServerS2A {
+    private static final Logger logger = Logger.getLogger(HelloWorldServerS2A.class.getName());
+    private Server server;
+    private final int port;
+    private final String certChainFilePath;
+    private final String privateKeyFilePath;
+    private final String trustCertCollectionFilePath;
+
+    private HelloWorldServerS2A(int port,
+                             String certChainFilePath,
+                             String privateKeyFilePath,
+                             String trustCertCollectionFilePath) {
+        this.port = port;
+        this.certChainFilePath = certChainFilePath;
+        this.privateKeyFilePath = privateKeyFilePath;
+        this.trustCertCollectionFilePath = trustCertCollectionFilePath;
+    }
+
+    private SslContextBuilder getSslContextBuilder() {
+      SslContextBuilder sslClientContextBuilder = SslContextBuilder.forServer(new File(certChainFilePath), new File(privateKeyFilePath));
+        if (trustCertCollectionFilePath != null) {
+            sslClientContextBuilder.trustManager(new File(trustCertCollectionFilePath));
+            sslClientContextBuilder.clientAuth(ClientAuth.REQUIRE);
+            sslClientContextBuilder.protocols("TLSv1.3");
+        }
+        return GrpcSslContexts.configure(sslClientContextBuilder);
+    }
+
+    private void start() throws IOException {
+        server = NettyServerBuilder.forPort(port)
+                .sslContext(getSslContextBuilder().build())
+                .addService(new GreeterImpl())
+                .build()
+                .start();
+
+        logger.info("Server started, listening on " + port);
+        Runtime.getRuntime().addShutdownHook(new Thread() {
+            @Override
+            public void run() {
+                // Use stderr here since the logger may have been reset by its JVM shutdown hook.
+                System.err.println("*** shutting down gRPC server since JVM is shutting down");
+                HelloWorldServerS2A.this.stop();
+                System.err.println("*** server shut down");
+            }
+        });
+    }
+
+    private void stop() {
+        if (server != null) {
+            server.shutdown();
+        }
+    }
+
+    /**
+     * Await termination on the main thread since the grpc library uses daemon threads.
+     */
+    private void blockUntilShutdown() throws InterruptedException {
+        if (server != null) {
+            server.awaitTermination();
```

Comment:
Can we just move this to `main` and get rid of this method?

Reply:
server is not static -- can't be referenced in static main method